### ISSUE TITLE
Update 8.1 instructions to use ELRepo

### ISSUE THREAD

**Issue by sethgoldin** [sethgoldin/install-davinci-resolve-centos#31]
ELRepo [has now packaged the NVIDIA drivers for 8.1](https://elrepo.org/bugs/view.php?id=944#c6874). Time to update the 8.1 guide to use ELRepo instead of NVIDIA's `.run` file, since the ELRepo packages are much easier to deal with.

Will update the guide as soon as I can physically get access to a workstation. Delayed right now from a COVID-19 lockdown.

**Comment by keksOS** [comment]
> ELRepo [has now packaged the NVIDIA drivers for 8.1](https://elrepo.org/bugs/view.php?id=944#c6874). Time to update the 8.1 guide to use ELRepo instead of NVIDIA's `.run` file, since the ELRepo packages are much easier to deal with.
> 
> Will update the guide as soon as I can physically get access to a workstation. Delayed right now from a COVID-19 lockdown.

This is something I am wondering how would one now go about updating the Drivers or going back to earlier driver - would I have to follow the steps again so I don't brake the boot ?  Thanks!  I think I am seeing a mem leak with the driver in your guide.  So I wanted to try the latest or go back one driver release but I don't want to brake my install :)

**Comment by sethgoldin** [comment]
Unfortunately I'm not quite sure how NVIDIA's `.run` file may interact with the drivers that are packaged from ELRepo.

What I would try is:
Try a clean install from the [8.1 instructions](https://github.com/sethgoldin/install-davinci-resolve-centos/blob/master/centos-8.1.md), but instead of steps 9 and 10 in the 8.1 instructions, substitute in steps 9, 10, and 11 from the [7.8 instructions](https://github.com/sethgoldin/install-davinci-resolve-centos/blob/master/centos-7.8.md). I'm guessing that `kmod-nvidia.x86_64` will call 440.82 for EL8.

**Comment by keksOS** [comment]
i have installed new driver version but following original guide did not want to trust ELRepo yet :)  But I am sad that windows 10 ent is a lot faster on standard candle test under lighter loads.  
GPU 2080ti
9 blur nodes - win10ent     80fps | centos8.1 - 64fps
18 blur nodes- win10ent    47fps | centos8.1 - 44fps
30 blur nodes- win10ent    31fps | centos8.1 - 30fps
66 blur nodes- win10 ent   15fps | centos8.1 - 16fps

1 TNR - win10ent    78fps | centos 8.1 - 64fps
2 TNR - win10ent    51fps | centos8.1 - 44fps
4 TNR - win10ent    28fps | centos8.1 - 27fps
6 TNR -win10ent     20fps | centos8.1 - 20fps

I have also noticed that the whole resolve GUI seems a lot more responsive under windows10 enterprise.  So I am questioning if Linux is worth it after all.

**Comment by sethgoldin** [comment]
This is very interesting, though I’ve always been somewhat skeptical of the [external validity](https://en.wikipedia.org/wiki/External_validity?wprov=sfti1) of the Standard Candle Test. Real-world workflows involve much more than just NR nodes!

Your results are plausible, but I’m curious to hear more about exact hardware and exact versions of each OS, NVIDIA driver, and versions of Resolve.

Also, did you try playing with [Tuned](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/monitoring_and_managing_system_status_and_performance/getting-started-with-tuned_monitoring-and-managing-system-status-and-performance)? For rendering, I like `throughput-performance`.

There are other reasons to use CentOS as well: stability, cost, and customization. Time savings from slightly increased render speeds can be totally cancelled out when the Windows Registry gets completely corrupt and requires a full system reset. 🙂

**Comment by keksOS** [comment]
HP Z800 workstation with dual Xeons and 48GB ram with 2080ti and Decklink Mini 4K Monitor running 2 x 30 inch Apple Cinema displays and Sony grading monitor. 3 x RAID 0 volumes running at around 1000MB/sec on LSI cards.  Both Win amd Centos8.1 had latest version of Resolve 16 and Nvidia drivers / cuda.  I also tested latest Ubuntu Studio, Manjaro and now I am installing BM 7.3 iso from a DVD drive.  Fastest and most fluid of all linux distros was Manjaro for now.  Ubuntu studio came close too. Still Windows is fastest and the most fluid UI also.  I played with many kernels in Manjaro using realtime kernels and different versions and cpu settings but did not notice anything different.  Some say to try ClearLinux as the fastest one but my z800 workstation does not have UEFI so I would need Clover EFI emulator to boot to install.

**Comment by sethgoldin** [comment]
Does Manjaro beat Windows in terms of performance? I wonder if merely using newer kernels makes the difference. For that reason, I'd be curious to see how Fedora might perform.

**Comment by keksOS** [comment]
Looks like Manjaro is now the fastest distro I tested.  I have installed it now on a bigger SSD and will stick with it.  I was able to match the speed of windows 10 ent with it only by closing the scopes on playback.  But on heavier loads it looks like its on par with windows give and take few frames.  One thing I have noticed is that the compositor for the desktop environment plays a crucial role in how fluid the interface seems.  I am now on the new KDE Plasma and I have set the OpenGL 3.1 for Rendering backend.  It is really fluid - the worst was actually CentOS8.1 - Today I ran Blackmagic Distro 7.3 iso and installed new drivers for 2080ti , Mini monitor 4K and it was running ok but not the best.  The GUI was not as fluid as Manjaro KDE Plasma but it was a lot better then CentOS8.1.  Manjaro also has a script you can build everything with one command with AUR https://aur.archlinux.org/packages/davinci-resolve-studio  in terminal or just by using GUI Add/Remove Software - it is a one click solution - same way with Decklink drivers - https://aur.archlinux.org/packages/decklink

What I like so far is that I can change and test different kernels and kernel-heathers all inside GUI System Settings and when adding new heathers in terminal you can do it to multiple kernels and it automatically finds DKMS and binds it to hardware drivers - like for instance decklink card.  Really impressive integration so far.   But Windows 10 enterprise still has a edge a bit.  I will do more project testing with 8K RED media and compare what I get as far as real world usage and how it feels.  I ran some quick tests now with 8K media and it was realtime but will have to do a comparing against win10 to see max fps possible under different loads.

I have a feeling that with multiple 2080ti cards Linux would be better but needs to be tested :)

I want to try next Clear Linux it is as I understand run by Intel Corporation and it has a cool way to keep the stacks clean as you update as to not loose performance vs security.  It should be the fastest linux and optimized for intel cpus.

**Comment by sethgoldin** [comment]
Time to check 16.2.3: https://www.blackmagicdesign.com/support/readme/ca91eb636f944eb2b7ff65ee56db41e5 😉

**Comment by keksOS** [comment]
Installed 16.2.3 on BM CentOS7.3 looks like it works little bit faster under heavy load now then win10ent but still slower on lighter loads.  Difference is not so big now and started to level out.  Next is to test different versions of drivers on CentOS7.3.

**Comment by sethgoldin** [comment]
I'd be more curious about a fresh installation of 7.8 from upstream CentOS. Since you're a bit more of an advanced user, you don't have to be stuck with the very old kernels that BMD shipped for their 7.3 ISO. Presumably there are performance optimizations in the newer kernels from 7.8.

**Comment by keksOS** [comment]
I am not sure we will see any gains because I tested 8.1 first but I can try it.  I think the main factor here are studio drivers on win10 vs linux.  Looks like linux drives are limited. What about HDR support and 30bit per channel that they added to studio drivers for RTX making them as Quadro in that redpect?   I have just started to dig deeper and as I see someone is making patches for linux drives. Lots of stuff is missing as I understand.
